editoast/src/views/sub_categories.rs

```diff
@@ -0,0 +1,151 @@
+use axum::Extension;
+use axum::extract::Json;
+use axum::extract::Path;
+use axum::extract::Query;
+use axum::extract::State;
+use axum::response::IntoResponse;
+use editoast_authz as authz;
+use editoast_derive::EditoastError;
+use editoast_models::DbConnectionPoolV2;
+use editoast_schemas::rolling_stock::SubCategory;
+use editoast_schemas::rolling_stock::TrainCategory;
+use serde::Deserialize;
+use serde::Serialize;
+use thiserror::Error;
+use utoipa::IntoParams;
+use utoipa::ToSchema;
+
+use crate::error::Result;
+use crate::views::AuthenticationExt;
+use crate::views::AuthorizationError;
+use crate::views::pagination::PaginationQueryParams;
+use crate::views::pagination::PaginationStats;
+
+crate::routes! {
+    "/sub_category" => {
+        get_sub_categories,
+        create_sub_categories,
+        "/{code}" => {
+            delete_sub_category,
+        },
+    },
+}
+
+editoast_common::schemas! {
+    SubCategoryPayload,
+    SubCategoriesCountList,
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "sub_categories")]
+pub enum SubCategoryError {
+    #[error(transparent)]
+    #[editoast_error(status = 500)]
+    Database(#[from] editoast_models::model::Error),
+}
+
+#[derive(Debug, Serialize, ToSchema)]
+struct SubCategoriesCountList {
+    results: Vec<SubCategory>,
+    #[serde(flatten)]
+    stats: PaginationStats,
+}
+
+#[utoipa::path(
+    get, path = "",
+    tag = "sub_categories",
+    responses(
+        (status = 200, description = "The list of all sub categories", body = SubCategoriesCountList),
+    ),
+)]
+async fn get_sub_categories(
+    State(_db_pool): State<DbConnectionPoolV2>,
+    Extension(auth): AuthenticationExt,
+    Query(_pagination): Query<PaginationQueryParams<100>>,
```

Comment:
```suggestion
    Query(_pagination): Query<PaginationQueryParams<1000>>,
```

we can probably go higher since the objects are lightweight